ArduPlane/commands.cpp

```diff
@@ -122,7 +122,13 @@ void Plane::update_home()
     }
     if (ahrs.home_is_set() && !ahrs.home_is_locked()) {
         Location loc;
-        if(ahrs.get_position(loc)) {
+        if(ahrs.get_position(loc) && gps.status() >= AP_GPS::GPS_OK_FIX_3D) {
+            // we take the altitude directly from the GPS as we are
+            // about to reset the baro calibration. We can't use AHRS
+            // altitude or we can end up perpetuating a bias in
+            // altitude, as AHRS alt depends on home alt, which means
+            // we would have a circular dependency
+            loc.alt = gps.location().alt;
```

Comment:
We purposefully took this away. This leads to the vehicle resetting altitude, but still reporting a non zero altitude on the HUD. At least make this an option, as this isn't remotely a problem if GPS is the primary reference into the EKF.

Reply:
actually it is a problem with EK2_ALT_SOURCE=2, as the backup AHR2 is still affected. I have a log showing that being off by 35m after sitting on the ground for a while. So if it does fallback to AHRS then it flies at totally the wrong altitude

Reply:
We need to find a better solution then, the HUD showing the wrong altitude while sitting on the ground is both a UI issue, and indicative of the fact that we do descent rate management wrong on quadplane for example.

Reply:
note that this issue has nothing to do with baro drift in the normal sense. There can be zero baro drift (just small noise centered on zero) and it can still end up having home off by many 10s of meters